daos-cortx/src/daos_kv_benchmark/kv_benchmarking.cc

```diff
@@ -0,0 +1,400 @@
+
+/* buffer to hold keys while listing */
+#define KEY_DESC_BUF (1024 * 1024)
+
+#define ARG_KEY_SIZE_OPTIONS            { BM_KEY_64B , BM_KEY_128B, BM_KEY_256B, BM_KEY_512B, BM_KEY_1024B }
+#define ARG_VAL_SIZE_OPTIONS            { BM_VAL_1K , BM_VAL_4K, BM_VAL_8K, BM_VAL_16K, BM_VAL_32K }
+#define NR_KEYS_PER_KV_SIZE_COMBINATION { NR_KEYS_100 /*,NR_KEYS_1000, NR_KEYS_10000, NR_KEYS_100000 */ }
+
+#define ARG_MATRICS\
+   { ARG_KEY_SIZE_OPTIONS, ARG_VAL_SIZE_OPTIONS, NR_KEYS_PER_KV_SIZE_COMBINATION }
+
+static char node[ 128 ] = "new_node";
+
+static daos_handle_t poh;
+static daos_handle_t coh;
+
+#define FAIL( fmt, ... )                                            \
+    do {                                                            \
+        fprintf(stderr, "Process (%s): " fmt " aborting\n",         \
+                node, ## __VA_ARGS__);                              \
+        exit(1);                                                    \
+    } while (0)
+
+#define ASSERT( cond, ... )                                         \
+    do {                                                            \
+        if (!(cond))                                                \
+        FAIL(__VA_ARGS__);                                          \
+    } while (0)
+
+#define BUFLEN 100
+
+daos_handle_t oh;
+daos_obj_id_t oid;
+int           rc;
+
+uuid_t pool_uuid, co_uuid;
+
+int setup_main( )
+{
+   /** initialize DAOS by connecting to local agent */
+   rc = daos_init( );
+   ASSERT( rc == 0, "daos_init failed with %d", rc );
+
+   rc = uuid_parse( POOL_ID, pool_uuid );
+
+   rc = daos_pool_connect( pool_uuid, NULL, DAOS_PC_RW, &poh,
+                           NULL, NULL );
+   ASSERT( rc == 0, "pool connect failed with %d", rc );
+
+   /** generate uuid for container */
+   uuid_generate( co_uuid );
+
+   /** create container */
+   rc = daos_cont_create( poh, co_uuid, NULL /* properties */,
+                          NULL /* event */ );
+   ASSERT( rc == 0, "container create failed with %d", rc );
+
+   /** open container */
+   rc = daos_cont_open( poh, co_uuid, DAOS_COO_RW, &coh, NULL,
+                        NULL );
+   ASSERT( rc == 0, "container open failed with %d", rc );
+
+   /** share container handle with peer tasks */
+
+   oid.hi = 0;
+   oid.lo = 4;
+
+   /** the KV API requires the flat feature flag be set in the oid */
+   daos_obj_generate_id( &oid, DAOS_OF_KV_FLAT, OC_SX, 0 );
+
+   rc = daos_kv_open( coh, oid, DAOS_OO_RW, &oh, NULL );
+   ASSERT( rc == 0, "KV open failed with %d", rc );
+}
+
+void tear_down( ) {
+   //close object handle
+   daos_kv_close( oh, NULL );
+
+   rc = daos_cont_close( coh, NULL );
+   ASSERT( rc == 0, "cont close failed" );
+
+   rc = daos_pool_disconnect( poh, NULL );
+   ASSERT( rc == 0, "disconnect failed" );
+
+   /** teardown the DAOS stack */
+   rc = daos_fini( );
+   ASSERT( rc == 0, "daos_fini failed with %d", rc );
+}
+
+void gen_key_names( char *key_buf, int nr, int key_size )
```

Comment:
Add comment about key_name format generated by this method.

Reply:
done